noxfile.py

```diff
@@ -21,8 +21,7 @@
 
 @nox.session(python=python_versions)
 def tests(session):
-    session.install("-r", "requirements/tests.txt")
-    session.install(".")
+    session.install(".[dev,tests]")
```

Comment:
The changes like these allow us to bypass the need for having the requirements files specified in the `requirements/` directory.  We'll be installing the dependencies for running tests based on the dependency sets in `pyproject.toml` instead, so dependencies will only need to be included in one place.